src/libs/ajax.js

```diff
@@ -792,6 +803,25 @@ const Workspaces = signal => ({
 })
 
 
+const DataRepo = signal => ({
+  snapshots: () => {
```

Comment:
This level of indirection isn't needed, because it's not capturing any variables the way `snapshot` below is. I think we should collapse this level to make it more concise.